pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -437,13 +439,20 @@ func (s *Service) runInstance(role string, i *v1alpha1.Instance) (*v1alpha1.Inst
 		return nil, errors.Errorf("no instance returned for reservation %v", out.GoString())
 	}
 
-	s.scope.V(2).Info("Waiting for instance to run", "instance-id", *out.Instances[0].InstanceId)
-	err = s.scope.EC2.WaitUntilInstanceRunningWithContext(
-		aws.BackgroundContext(),
+	waitTimeout := 1 * time.Minute
```

Comment:
Aren't `WaiterOptions` applied across every aws sdk call though? I mentioned above I'm open to actually remove this call entirely, we can rely on reconciliation to go and describe the state in a later reconciliation

Reply:
iirc, we originally added this because we had hit issues with returning without waiting previously. I want to say it might have been related to how we check for instance availability during `Exists`.

Reply:
I think that has been fixed by adding the "pending" checks to Exists https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/aws/services/ec2/instances.go#L51

Reply:
These don't apply across every sdk call, only for the waiter that gets created with the call.

Reply:
Ah ok, I was looking at something else